paasta_tools/paasta_service_config_loader.py

```diff
@@ -115,7 +116,10 @@ def instance_configs(
         if (cluster, instance_type_class) not in self._framework_configs:
             self._refresh_framework_config(cluster, instance_type_class)
         for instance, config in self._framework_configs.get((cluster, instance_type_class), {}).items():
-            yield self._create_service_config(cluster, instance, config, instance_type_class)
+            try:
+                yield self._create_service_config(cluster, instance, config, instance_type_class)
+            except NoDeploymentsAvailable:
```

Comment:
This is kinda a departure from how we used to do it, where all the callers *must* catch NoDeployments and skip. 
I like the idea of never even bothering to show them this (pass). It is very rare that we are scratching our heads wondering why something is not happening because it is not deployed. (but it happened earlier when paasta was in its infancy)

Reply:
I'd prefer to keep the exception handling in the caller, but there's no clean way to do it with a generator like this. If you raise an exception, then any instances that would come afterwards will never get yielded.

I _could_ do something like yielding functions, which either return the instance config or raise NoDeploymentsAvailable, but I think that'd be a weird interface.